binary_search_tree/tree.py

```diff
@@ -14,42 +14,120 @@ class Tree:
+
+    def find_helper(self, current_node, key):
+        if current_node == None:
+            return None
+        elif current_node.key == key:
+            return current_node.value
+        elif current_node.key > key:
+            return self.find_helper(current_node.left, key)
+        else:
+            return self.find_helper(current_node.right, key)
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(1) 
     def find(self, key):
-        pass
+        if self.root == None:
+            return None
 
-    # Time Complexity: 
-    # Space Complexity: 
+        return self.find_helper(self.root, key)
+        
+
+    def inorder_helper(self, current_node, items):
+        if current_node is not None:
+            self.inorder_helper(current_node.left, items)
+            items.append({"key": current_node.key, "value":current_node.value})
+            self.inorder_helper(current_node.right, items)
+
+    # Time Complexity:O(n) 
+    # Space Complexity: O(n)
     def inorder(self):
-        pass
+        items = []
+
+        self.inorder_helper(self.root, items)
+        return items
 
-    # Time Complexity: 
-    # Space Complexity:     
+    def preorder_helper(self, current_node, items):
+        if current_node is not None:
+            items.append({"key": current_node.key, "value":current_node.value})
+            self.preorder_helper(current_node.left, items)
+            self.preorder_helper(current_node.right, items)
+        
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)  
     def preorder(self):
-        pass
+        items = []
+        self.preorder_helper(self.root, items)
+        return items
+
+
+    def postorder_helper(self, current_node, items):
+        if current_node is not None:
+            self.postorder_helper(current_node.left, items)
+            self.postorder_helper(current_node.right, items)
+            items.append({"key":current_node.key, "value": current_node.value})
 
-    # Time Complexity: 
-    # Space Complexity:     
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)  
     def postorder(self):
-        pass
+        items = []
+        self.postorder_helper(self.root, items)
+        return items
+
+    # Time Complexity:O(n) 
+    # Space Complexity: O(n) 
+
+    def height_helper(self, current_node):
+        if current_node is None:
+            return 0
+        left_height = self.height_helper(current_node.left)
+        right_height = self.height_helper(current_node.right)
+        return max(left_height, right_height) + 1
+
 
-    # Time Complexity: 
-    # Space Complexity:     
     def height(self):
-        pass
+        if self.root is None:
+            return 0
+            
+        return self.height_helper(self.root)
 
 
 #   # Optional Method
-#   # Time Complexity: 
-#   # Space Complexity: 
+#   # Time Complexity: O(n^2)
+#   # Space Complexity: O(n)
     def bfs(self):
```

Comment:
✨Nice! However, time complexity is O(n) for this. I would also recommend not nesting the helper function when your other helper functions aren't nested!